From d7c02a69eb989b205888fe0305c244305cca9e62 Mon Sep 17 00:00:00 2001
From: Danny Lin <danny@kdrag0n.dev>
Date: Tue, 11 Aug 2026 09:47:53 +0000
Subject: [PATCH] SystemUI: Always refresh power menu on UI mode change

This is necessary for reliably theming the global actions dialog with
dynamic colors and adapting it to light/dark themes.

Demo screenshots (with color overlays applied):
https://twitter.com/kdrag0n/status/1445960685427433473

[kdrag0n: ported to 12L]

Change-Id: If58fb4079a4cd11414ff928fad3576beecb14ff5
Signed-off-by: althafvly <althafvly@gmail.com>
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>

[Port to A16] LOS23 replaced global_actions_lite_* hardcoded colors with theme-attribute drawables (materialColorSurfaceContainerLow etc.), so the colors.xml parts are obsolete; kept the onUiModeChanged refresh hook.

Change-Id: Ia8090c63bc47692091be7c21a4534276b15658ae
---
 .../systemui/globalactions/GlobalActionsDialogLite.java  | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialogLite.java b/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialogLite.java
index 528e7d871ac5..55aa30289941 100644
--- a/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialogLite.java
+++ b/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialogLite.java
@@ -968,6 +968,15 @@ public class GlobalActionsDialogLite implements DialogInterface.OnDismissListene
                 user.id) != 0;
     }
 
+    @Override
+    public void onUiModeChanged() {
+        // Colors may change, depending on UI mode
+        mContext.getTheme().applyStyle(mContext.getThemeResId(), true);
+        if (mDialog != null && mDialog.isShowing()) {
+            mDialog.refreshDialog();
+        }
+    }
+
     @Override
     public void onConfigChanged(Configuration newConfig) {
         if (mDialog != null && mDialog.isShowing()
-- 
2.43.0

